s2wav/axisym_filters.py

```diff
@@ -0,0 +1,89 @@
+import numpy as np
+from s2wav import tiling
+from s2wav import helper_functions
+from typing import Tuple
+import pytest
+
+def k_lam(L:int, lam:float, quad_iters: int=300) -> float:
+    """Compute function k_lam defined in equation (12) in [1] used to compute scaling function generating function and
+        wavelet generating function.
+
+    Args:
+        L (int): Harmonic band-limit.
+        lam (float): Wavelet parameter which determines the scale factor between consecutive wavelet scales.
+        quad_iters (int): Total number of iterations for quadrature integration. Defaults to 300.
+    Returns:
+        np.ndarray: value of k_lam computed for values between 1/lam and 1, parametrised by l as required to compute 
+        the axisymmetric filters in tiling_axisym()
+
+    Notes:
+        [1] B. Leidstedt et. al., "S2LET: A code to perform fast wavelet analysis on the sphere", A&A, vol. 558, p. A128, 2013.
+    """
+
+    J = helper_functions.j_max(L, lam)
+
+    normalisation = tiling.part_scaling_fn(1/lam, 1.0, quad_iters, lam)
+    k = np.zeros((J + 2) * L)
+    
+
+    for j in range(J+2):
+        for l in range(L):
+            if l < lam**(j-1):
+                k[l + j * L] = 1
+            elif l > lam**j:
+                k[l + j * L] = 0
+            else:
+                k[l + j * L] =  tiling.part_scaling_fn(l/lam**j, 1.0, quad_iters, lam)/ normalisation
+            
+
+    return k
+
+
+def filters_axisym(L: int, lam:float, J_min: int) -> Tuple[np.ndarray]:
+    """Computes wavelet kernels and scaling kernel in harmonic space, according to equations (15), (16) in [1]
+        without the normalisation factor.
+    
+    Args:
+        L (int): Harmonic band-limit.
+        lam (float): Wavelet parameter which determines the scale factor between consecutive wavelet scales.
+        J_min (int): First wavelet scale used.
+    Returns:
+        Tuple[np.ndarray]: Unnormalised wavelet kernels and scaling kernel in harmonic space.
```

Comment:
I'd recommend adding a couple of extra lines below defining what phi and psi are.

Reply:
I've added a few words to describe these things.